### ISSUE TITLE
no-shadow if const and enum value are the same

### ISSUE THREAD

**Issue by samijaber** [eslint/typescript-eslint-parser#552]
**What version of TypeScript are you using?**
3.1.6

**What version of `typescript-eslint-parser` are you using?**
21.0.1

**What code were you trying to parse?**

```ts
const f = 0;
enum Foo {
  f,
}
```

**What did you expect to happen?**
No errors

**What happened?**
Got a `no-shadow` error

Might be somewhat related to #459

**Comment by mysticatea** [comment]
Thank you for this report.

However, this is an expected behavior because the enum member `f` literally shadows the upper `f`.

```ts
const f = 777;
enum E1 {
    f = 1, // Shadows the upper `f`.
    g = f, // = 1
}
enum E2 {
    f1 = 1,
    g = f, // This is the upper `f`.
}
```

> [Open playground](https://www.typescriptlang.org/play/index.html#src=const%20f%20%3D%20777%3B%0D%0Aenum%20E1%20%7B%0D%0A%20%20%20%20f%20%3D%201%2C%20%2F%2F%20Shadows%20the%20upper%20%60f%60.%0D%0A%20%20%20%20g%20%3D%20f%2C%20%2F%2F%20%3D%201%0D%0A%7D%0D%0Aenum%20E2%20%7B%0D%0A%20%20%20%20f1%20%3D%201%2C%0D%0A%20%20%20%20g%20%3D%20f%2C%20%2F%2F%20This%20is%20the%20upper%20%60f%60.%0D%0A%7D%0D%0A)
